src/components/MentorDetails.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Link, useParams } from 'react-router-dom';
+import { fetchMentorDetails } from '../redux/slices/mentors/mentorDetailsSlice';
+
+const MentorDetails = () => {
+  const { id } = useParams();
+  const dispatch = useDispatch();
+  const { details, loading, error } = useSelector(
+    (state) => state.mentorDetails,
+  );
+
+  useEffect(() => {
+    dispatch(fetchMentorDetails(id));
+  }, [dispatch, id]);
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  if (error) {
+    return (
+      <div>
+        Error:
+        {error}
+      </div>
+    );
+  }
```

Comment:
Please refactor the code to utilize the `ErrorStatus` and `LoadingStatus` components designed specifically for displaying the loading state and error messages within the application. These components are located in the `src/components/ux/` directory.

Example: 

```jsx
{loading && <LoadingStatus />}
{error && <ErrorStatus error={error} />}
```